pytest_odoo.py

```diff
@@ -126,109 +390,137 @@ def enable_odoo_test_flag():
-            )
-        self.config.pluginmanager.consider_module(mod)
-        return mod
-
-    def __repr__(self):
-        return "<Module %r>" % (getattr(self, "name", None), )
+    def _getobj(self):
+        return importtestmodule(self.path, self.config)
 
 
-class OdooTestPackage(_pytest.python.Package, OdooTestModule):
+class OdooTestPackage(Package):
     """Package with odoo module lookup.
 
     Any python module inside the package will be imported with
     the prefix `odoo.addons`.
 
     This class is used to prevent loading odoo modules in duplicate,
     which happens if a module is loaded with and without the prefix.
+
+    Original at https://github.com/pytest-dev/pytest/blob/6bd3f313447290380cbc2db30fb9ee5cca7eb941/src/_pytest/python.py#L619
     """
 
-    def __repr__(self):
-        return "<Package %r>" % (getattr(self, "name", None), )
+    def setup(self) -> None:
+        init_mod = importtestmodule(self.path / "__init__.py", self.config)
+
+        # Not using fixtures to call setup_module here because autouse fixtures
+        # from packages are not called automatically (#4085).
+        setup_module = _get_first_non_fixture_func(
+            init_mod, ("setUpModule", "setup_module")
+        )
+        if setup_module is not None:
+            _call_with_optional_argument(setup_module, init_mod)
+
+        teardown_module = _get_first_non_fixture_func(
+            init_mod, ("tearDownModule", "teardown_module")
+        )
+        if teardown_module is not None:
+            func = partial(_call_with_optional_argument, teardown_module, init_mod)
+            self.addfinalizer(func)
 
 
+@pytest.hookimpl()
 def pytest_pycollect_makemodule(module_path, path, parent):
     if not _find_manifest_path(module_path):
         return None
-    if path.basename == "__init__.py":
-        return OdooTestPackage.from_parent(parent, path=Path(path))
-    else:
-        return OdooTestModule.from_parent(parent, path=Path(path))
+    return OdooTestModule.from_parent(parent, path=Path(path))
 
 
-def _find_manifest_path(collection_path: Path) -> Path:
+@pytest.hookimpl()
+def pytest_collect_directory(path, parent):
+    if not _find_manifest_path(path):
+        return None
+    pkginit = path / "__init__.py"
+    if pkginit.is_file():
+        return OdooTestPackage.from_parent(parent, path=path)
+    return None
+
+
+def _find_manifest_path(collection_path: Path) -> Path | None:
     """Try to locate an Odoo manifest file in the collection path."""
     # check if collection_path is an addon directory
     path = collection_path
     level = 0
-    while level < 5  and not (path.parent / "__manifest__.py").is_file():
+    while level < 5 and not (path / "__manifest__.py").is_file():
         path = path.parent
         level += 1
-    return path.parent / "__manifest__.py"
+    if not (path / "__manifest__.py").is_file():
+        return None
+    return path / "__manifest__.py"
```

Comment:
In #70 I've fixed that a bit differently and add unit test for it, do you mind to at least add the unit test?

Reply:
as far I remember problem, using pytest8 this method return `/__manifest__.py` (or `/some/path/__manifest__.py` it starting point is deeper than 5 levels) even the file does not exists on the file system. (I'm wondering why it's not a matter in pytest v7 !